libpod/container_internal_common.go

```diff
@@ -177,6 +179,29 @@ func getOverlayUpperAndWorkDir(options []string) (string, string, error) {
 	return upperDir, workDir, nil
 }
 
+var (
+	hasCapSysResourceOnce sync.Once
+	hasCapSysResourceRet  bool
+	hasCapSysResourceErr  error
+)
+
+// hasCapSysResource returns whether the current process has CAP_SYS_RESOURCE.
+func hasCapSysResource() (bool, error) {
+	hasCapSysResourceOnce.Do(func() {
+		currentCaps, err := capability.NewPid2(0)
+		if err != nil {
+			hasCapSysResourceErr = err
+			return
+		}
+		if err = currentCaps.Load(); err != nil {
+			hasCapSysResourceErr = err
+			return
+		}
+		hasCapSysResourceRet = currentCaps.Get(capability.EFFECTIVE, capability.CAP_SYS_RESOURCE)
+	})
+	return hasCapSysResourceRet, hasCapSysResourceErr
+}
```

Comment:
Nit: that works fine to cache but a bit more modern is to use sync.OnceValues(), that way you do not need to declare multiple global vars and just one function where the vars are in the local scope which makes it more readable IMO.

Reply:
thanks, fixed and pushed a new version